src/Servers/HttpSys/src/RequestProcessing/RequestStream.cs

```diff
@@ -190,111 +190,10 @@ internal void UpdateAfterRead(uint statusCode, uint dataRead)
         }
 
         public override unsafe IAsyncResult BeginRead(byte[] buffer, int offset, int size, AsyncCallback? callback, object? state)
-        {
-            ValidateReadBuffer(buffer, offset, size);
```

Comment:
The async-version has the same code-path, so this removal should be OK in terms of correctness.

But with this change it may allocate more.
The async version needs to allocate a `RequestStreamAsyncResult` too, plus there's the `Task`-object.

I don't have any feeling if / how this method is called and used, so this may not be a concern at all.

But while looking at the code for `RequestStreamAsyncResult` I saw the TCS, which could replaced by a non-allocating variant --> https://github.com/gfoidl/aspnetcore/commit/e11e537c49fee94c4abcaf656841e2f34c6e7645
Is this something to pursue? (Maybe in a separate issue / PR).

Reply:
Removing the redundant code is likely for the best.

Feel free to follow-up about the TCS in a different PR.